lldb/source/Plugins/LanguageRuntime/Swift/SwiftLanguageRuntimeDynamicTypeResolution.cpp

```diff
@@ -1844,29 +1949,65 @@ bool SwiftLanguageRuntimeImpl::GetDynamicTypeAndAddress_Protocol(
   if (log)
     log->Printf("existential address is 0x%llx", existential_address);
 
-  if (!existential_address || existential_address == LLDB_INVALID_ADDRESS)
+  if (!existential_address || existential_address == LLDB_INVALID_ADDRESS) {
+    if (log)
+      log->Printf("Existential address is invalid");
     return false;
+  }
+
+  // If the protocol list contains any types declared in LLDB,
+  // we use the remote AST implementation since we currently
+  // don't register the metadata generated by LLDB.
+  if (CheckIfAnyTypesDeclaredInLLDB(protocol_typeref)) {
+    auto pair = remote_ast_impl(use_local_buffer, existential_address);
+    if (!pair)
+      return false;
+    class_type_or_name.SetCompilerType(std::get<CompilerType>(*pair));
+    address = std::get<Address>(*pair);
+    return true;
+  }
 
   if (use_local_buffer)
     PushLocalBuffer(existential_address, in_value.GetByteSize().getValueOr(0));
 
   swift::remote::RemoteAddress remote_existential(existential_address);
-  auto result = remote_ast.getDynamicTypeAndAddressForExistential(
-      remote_existential, GetSwiftType(protocol_type));
 
+  auto *reflection_ctx = GetReflectionContext();
+  auto pair = reflection_ctx->projectExistentialAndUnwrapClass(
+      remote_existential, *protocol_typeref);
   if (use_local_buffer)
     PopLocalBuffer();
 
-  if (!result.isSuccess()) {
+  if (!pair) {
     if (log)
-      log->Printf("RemoteAST failed to get dynamic type of existential");
+      log->Printf("Runtime failed to get dynamic type of existential");
     return false;
   }
 
-  auto type_and_address = result.getValue();
-  class_type_or_name.SetCompilerType(
-      ToCompilerType(type_and_address.InstanceType));
-  address.SetRawAddress(type_and_address.PayloadAddress.getAddressData());
+  const swift::reflection::TypeRef *typeref = nullptr;
```

Comment:
I would probably leave out the = nullptr here and on the next line, since the initialization *right* below.